Index: backports-6.12.52/net/mac80211/tx.c
===================================================================
--- backports-6.12.52.orig/net/mac80211/tx.c
+++ backports-6.12.52/net/mac80211/tx.c
@@ -5042,11 +5042,12 @@ static void ieee80211_set_beacon_cntdwn(
 
 static u8 __ieee80211_beacon_update_cntdwn(struct beacon_data *beacon)
 {
-	beacon->cntdwn_current_counter--;
 
 	/* the counter should never reach 0 */
 	WARN_ON_ONCE(!beacon->cntdwn_current_counter);
 
+	beacon->cntdwn_current_counter--;
+
 	return beacon->cntdwn_current_counter;
 }
 
@@ -5316,7 +5317,13 @@ ieee80211_beacon_get_ap(struct ieee80211
 	u16 csa_off_base = 0;
 	int mbssid_len;
 
-	if (beacon->cntdwn_counter_offsets[0]) {
+	bool short_beacon = (vif->bss_conf.dtim_period > 1);
+
+	if (ap->ps.dtim_count > 0)
+		short_beacon = ((ap->ps.dtim_count-1) != 0);
+
+	 /* Do not count channel switch count for short beacons */
+	if (beacon->cntdwn_counter_offsets[0] && !short_beacon) {
 		if (!is_template)
 			ieee80211_beacon_update_cntdwn(vif, link->link_id);
 
