From ff4f0893619b1c590f32e582b3a099480aac4ad0 Mon Sep 17 00:00:00 2001
From: Ria Thomas <ria.thomas@morsemicro.com>
Date: Thu, 13 Nov 2025 15:36:25 +0530
Subject: [PATCH] mac: Add RI config support and enable NDP BA for hw

---
 command.c        | 15 +++++++++++++++
 command.h        |  1 +
 mac.c            |  7 +++++++
 morse_commands.h | 20 ++++++++++++++++++++
 4 files changed, 43 insertions(+)

--- a/command.c
+++ b/command.c
@@ -620,3 +620,17 @@ int mm81x_cmd_get_disabled_channels(

 	return mm81x_cmd_tx(mm, (struct mm_cmd_resp *)resp, &req, resp_len, 0);
 }
+
+int mm81x_cmd_config_ri(struct mm81x *mm, u16 vif_id, s8 ri)
+{
+	struct mm_cmd_resp_set_response_indication resp;
+	struct mm_cmd_req_set_response_indication req = {
+		.response_indication = ri,
+	};
+
+	mm81x_cmd_init(mm, &req.hdr, MM_CMD_ID_SET_RESPONSE_INDICATION,
+			vif_id, sizeof(req));
+
+	return mm81x_cmd_tx(mm, (struct mm_cmd_resp *)&resp,
+			   (struct mm_cmd_req *)&req, sizeof(resp), 0);
+}
--- a/command.h
+++ b/command.h
@@ -102,5 +102,6 @@ int mm81x_cmd_set_frag_threshold(struct
 int mm81x_cmd_get_disabled_channels(
 	struct mm81x *mm, struct mm_cmd_resp_get_disabled_channels *resp,
 	uint resp_len);
+int mm81x_cmd_config_ri(struct mm81x *mm, u16 vif_id, s8 ri);

 #endif /* !_MM81X_COMMAND_H_ */
--- a/mac.c
+++ b/mac.c
@@ -1741,6 +1741,12 @@ static void mm81x_mac_ops_bss_info_chang
 			mm81x_err(mm, "mm81x_cmd_cfg_bss failed %d", ret);
 	}

+	if (changed & BSS_CHANGED_S1G_RI) {
+		ret = mm81x_cmd_config_ri(mm, mm_vif->id, info->s1g_ri);
+		if (ret)
+			mm81x_err(mm, "mm81x_cmd_config_ri failed %d", ret);
+	}
+
 	mutex_unlock(&mm->lock);
 }

@@ -2599,6 +2605,7 @@ static void mm81x_mac_config_hw(struct m
 	ieee80211_hw_set(hw, NEED_DTIM_BEFORE_ASSOC);
 	ieee80211_hw_set(hw, PS_NULLFUNC_STACK);
 	ieee80211_hw_set(hw, SUPPORTS_TX_FRAG);
+	ieee80211_hw_set(hw, SUPPORTS_NDP_BLOCKACK);

 	SET_IEEE80211_PERM_ADDR(hw, mm->macaddr);

--- a/morse_commands.h
+++ b/morse_commands.h
@@ -163,6 +163,7 @@ enum mm81x_cmd_id {
 	MM_CMD_ID_EVT_BEACON_FILTER_MATCH = 0x4016,

 	/* Test commands starting at 0x8000 */
+	MM_CMD_ID_SET_RESPONSE_INDICATION = 0x8007,
 	MM_CMD_ID_SET_TRANSMISSION_RATE = 0x8009,
 	MM_CMD_ID_SET_CAPABILITIES = 0x8118,

@@ -2992,6 +2993,25 @@ struct mm_cmd_resp_set_capabilities {
 	struct mm_cmd_header hdr;
 	__le32 status;
 } __packed;
+
+/**
+ * struct mm_cmd_req_set_response_indication - request message for SET_RESPONSE_INDICATION
+ *
+ * Test command for forcing the response indication field on TX packets. @note Valid values, are 0,
+ * 1, 2, 3. Set to -1 to disable this forcing of ri.
+ */
+struct mm_cmd_req_set_response_indication {
+	struct mm_cmd_header hdr;
+	s8 response_indication;
+} __packed;
+
+/**
+ * struct mm_cmd_resp_set_response_indication - response message for SET_RESPONSE_INDICATION
+ */
+struct mm_cmd_resp_set_response_indication {
+	struct mm_cmd_header hdr;
+	__le32 status;
+} __packed;

 /**
  * struct mm_cmd_req_set_transmission_rate - request message for SET_TRANSMISSION_RATE